lib/prism/ripper_compat.rb

```diff
@@ -238,13 +333,20 @@ def visit_number(node)
         value = yield slice[1..-1]
 
         bounds(node.location)
-        on_unary(RUBY_ENGINE == "jruby" ? :- : :-@, value)
+        on_unary(RUBY_ENGINE == "jruby" && Gem::Version.new(JRUBY_VERSION) < Gem::Version.new("9.4.6.0") ? :- : :-@, value)
```

Comment:
Could you make this something like:

```ruby
if RUBY_ENGINE == "jruby" && Gem::Version.new(JRUBY_VERSION) < Gem::Version.new("9.4.6.0")
  def visit_unary_operator(value)
    value == :-@ ? :- : value
  end
else
  def visit_unary_operator(value)
    value
  end
end
```

That way the check is only performed once.

Reply:
Sure. Pushed.